test/runnable/mixin1.d

```diff
@@ -730,27 +726,6 @@ void test29()
 
 /*******************************************/
 
-class A30
-{
-    template ctor(Type)
-    {
-        this(Type[] arr)
-        {
-            foreach(Type v; arr) writeln(typeid(typeof(v)));
```

Comment:
Isn't it just a matter of calling `toString` here ?

Reply:
I think for many that are `printf("%.*s\n", cast(int)darray.length, darray.ptr)`, better off using `puts()` instead, as dmd will handle D array->C pointer conversations for us.

Reply:
Is NUL-termination guaranteed tho ?

Reply:
For compile-time generated strings, the answer should always be yes.  I'm sure that there are some tests that exist to enforce it too (I know I've hit them in gdc code generation when I used to omit the trailing `\0`).